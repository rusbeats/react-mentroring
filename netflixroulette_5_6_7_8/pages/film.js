import React from 'react';
import fetch from 'isomorphic-fetch';
import { connect } from 'react-redux';
import { getUrlWithGenreFilter, getUrlWithId, isEmptyObject } from '../src/utils/helpers';
import { filmNotFound, receiveFilmById, receiveFilms } from '../src/store/actions';
import FilmDetailsPage from '../src/components/FilmDetailsPage';

export const Film = () => <FilmDetailsPage />;
//Homework 7
Film.getInitialProps = async ({ store, query }) => {
    try {
        const { id } = query;
        const { dispatch } = store;

        const fetchFilmUrl = getUrlWithId(id);
        const filmRes = await fetch(fetchFilmUrl);
        const filmData = await filmRes.json();

        if (isEmptyObject(filmData)) {
            dispatch(filmNotFound());
        } else {
            const genre = filmData.genres[0];
            const urlWithFilter = getUrlWithGenreFilter(genre);

            const filmsRes = await fetch(urlWithFilter);
            const { data } = await filmsRes.json();

            dispatch(receiveFilmById(filmData));
            dispatch(receiveFilms(data));
        }
    } catch (e) {
        console.error(e);
    }
};

export default connect(state => state)(Film);
