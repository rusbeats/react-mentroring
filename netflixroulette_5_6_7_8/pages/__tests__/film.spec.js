import React from 'react';
import { shallow } from 'enzyme';
import { Film } from '../film';

describe('Film', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(<Film />);
        expect(component).toMatchSnapshot();
    });
});
