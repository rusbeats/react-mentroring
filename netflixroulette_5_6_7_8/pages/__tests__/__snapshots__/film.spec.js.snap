// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film should render correctly (snapshot) 1`] = `<FilmDetailsPage />`;
